src/components/common/Extensions/ColonySwitcher/ColonySwitcher.tsx

```diff
@@ -52,69 +52,71 @@ const ColonySwitcher = () => {
-                  {
-                    'w-[26.75rem] border-none shadow-none': isMobile,
-                    'w-[15.1875rem]': !isMobile,
-                  },
-                ),
-              })}
-            >
-              {userLoading && (
-                <div className="h-[24.75rem] p-1 flex justify-center">
-                  <SpinnerLoader appearance={{ size: 'medium' }} />
-                </div>
-              )}
-              {!!watchlist.length && !userLoading && <ColoniesDropdown watchlist={[...watchlist].sort(sortByDate)} />}
-            </div>
-          )}
-          {isMobile && (
-            <ColonyDropdownMobile isOpen={isOpen} userLoading={userLoading}>
-              {!!watchlist.length && !userLoading && (
-                <ColoniesDropdown watchlist={[...watchlist].sort(sortByDate)} isMobile={isMobile} />
-              )}
-            </ColonyDropdownMobile>
-          )}
-        </div>
-      )}
-
-      {isMobile && isOpen && (
+    <div className="sm:mr-[2.5rem] mt-4 sm:mt-0">
+      <div className="flex justify-between relative" ref={ref}>
         <button
+          aria-label={formatMessage({ id: 'open.dropdown' })}
+          className={clsx('flex items-center justify-between', {
+            'w-[3.5225rem]': !isMobile,
+          })}
+          onClick={() => setIsOpen((prevState) => !prevState)}
           type="button"
-          aria-label={formatMessage({ id: 'close.dropdown' })}
-          onClick={() => setIsOpen(false)}
-          className="text-gray-400 pr-4"
         >
-          <Icon name="close" appearance={{ size: 'extraTiny' }} />
+          <ColonyAvatarWrapper
+            isOpen={isOpen}
+            isMobile={isMobile}
+            colonyToDisplay={colonyToDisplay}
+            colonyToDisplayAddress={colonyToDisplayAddress}
+            setTriggerRef={setTriggerRef}
+          />
         </button>
-      )}
+
+        {/* @TODO: add wallet buttons */}
+
+        {isOpen && (
+          <div className="h-auto absolute top-[8.1rem] sm:top-[2.3rem]">
+            {!isMobile && (
+              <div
+                ref={setTooltipRef}
+                {...getTooltipProps({
+                  className: clsx(
+                    `${styles.tooltipContainer} h-[24.75rem] p-1 flex justify-center z-[9999] tooltip-container`,
+                    {
+                      'w-[26.75rem] border-none shadow-none': isMobile,
+                      'w-[15.1875rem]': !isMobile,
+                    },
+                  ),
+                })}
+              >
+                {userLoading && (
+                  <div className="h-[24.75rem] p-1 flex justify-center">
+                    <SpinnerLoader appearance={{ size: 'medium' }} />
+                  </div>
+                )}
+                {!!watchlist.length && !userLoading && <ColoniesDropdown watchlist={[...watchlist].sort(sortByDate)} />}
```

Comment:
I see you use twice `[...watchlist].sort(sortByDate)}`
Maybe you can assign to the variable this one?